lib/rubocop/cop/infinum/factory_bot_association.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Infinum
+      # This cop looks for 'association' in factories that specify a ':factory' option
+      #
+      # @example
+      #   #bad
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'Lord of the Rings' }
+      #        association :author
+      #      end
+      #    end
+      #
+      #    FactoryBoy.define do
+      #      factory :book do
+      #        title {'Lord of the Rings'}
+      #        author { association :author }
+      #      end
+      #    end
+      #
+      #    #good
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'Lord of the Rings' }
+      #        author { build(:author) }
+      #      end
+      #    end
+      #
+      #    FactoryBot.define do
+      #      factory :author do
+      #        name { 'Ivan Gundulic' }
+      #      end
+      #    end
+      class FactoryBotAssociation < ::RuboCop::Cop::Cop
+        MSG = 'Use %<association_name>s { build(:%<factory_name>s) } instead'
+
+        def_node_matcher :association_matcher, <<~PATTERN
+          (send nil? :association (:sym $_) (hash (pair (sym :factory) (:sym $_))) ?)
+        PATTERN
+
+        def_node_matcher :association, <<~PATTERN
```

Comment:
```suggestion
        def_node_matcher :captured_association, <<~PATTERN
```

I favor this description because it's more explicit

Reply:
Thanks, was not sure what to call it, accidentally sent it this way, but wanted to ask you about naming it 😄